Release/v1.10.0

robkooper opened the pull request:
this should be release v1.10.0

Sweetdevil144 committed:
Update str_ns declaration

Signed-off-by: Abhinav Pandey <abhinav.pandey.met22@itbhu.ac.in>

Sweetdevil144 committed:
Update site$lon

Signed-off-by: Abhinav Pandey <abhinav.pandey.met22@itbhu.ac.in>

infotroph committed:
dbsync tweaks

- only cache geoip fields we use
- leaflet style names have changed
- merge instead of rbind to avoid duplicates
- typos

infotroph committed:
move NEWS into unreleased section

Commit:
Revert change.

Commit:
Revert change.

infotroph committed:
Apply suggestions from code review

infotroph committed:
Update base/db/R/convert_input.R

infotroph committed:
Merge pull request #3338 from Sweetdevil144/gsoc/convert-input

Refactor `convert_input` to Perform tasks via helper function

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

AritraDey-Dev committed:
fix spacing

Signed-off-by: Aritra Dey <adey01027@gmail.com>

Commit:
Add debias code.

infotroph committed:
Roxygen

infotroph committed:
manpage too

infotroph committed:
wording and whitespace

infotroph committed:
use settings hash as ensemble id if not provided

infotroph committed:
filter to pfts present at this site

infotroph committed:
pass site id when naming rundirs

Commit:
Move the previous debias script to the inst folder.

Commit:
remove previous debias functions.

Commit:
Add the main SDA debias workflow.

Commit:
Add the new debias workflow to the multi-site version.

Commit:
Add the new bias correction workflow to the sda_local script.

Commit:
Add the new SDA debiasing workflow to the SDA job submission function.

AritraDey-Dev committed:
feat(db): Re-enable PFT cultivar query test

Signed-off-by: Aritra Dey <adey01027@gmail.com>

Commit:
Move out of the if control.

Commit:
Bug fix.

Commit:
Bug fix

Commit:
Remove the condition for the wrapping where we have more than 20 characters.

infotroph committed:
Merge pull request #3651 from AritraDey-Dev/fix/reenable-db-test

feat(db): Re-enable PFT cultivar query test

Commit:
Update the runner script.

AritraDey-Dev committed:
test: note expected output length and specific variable checks

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
fix: api container failed to start due to issue in ma.R file

Signed-off-by: Aritra Dey <adey01027@gmail.com>

infotroph committed:
Merge pull request #3653 from AritraDey-Dev/fix/api-docker-container

fix: error in ma.R causing API container crashes

Commit:
remove arg.

Commit:
Update script.

Commit:
Revert "Move out of the if control."

This reverts commit 77bda481f7bc9a4065d8b05a16145ff03f4b2201.

Commit:
Resolve conflicts.

Commit:
Revert back.

infotroph committed:
changelog

infotroph committed:
Update CHANGELOG.md

dlebauer committed:
return NA if no validator

Co-authored-by: Chris Black <chris@ckblack.org>

dlebauer committed:
Update models/sipnet/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

dlebauer committed:
Update docker/depends/pecan_package_dependencies.csv

Co-authored-by: Chris Black <chris@ckblack.org>

dlebauer committed:
- require testthat >= 3.1.0 to support new mocking features in tests (3.1.0 was released in 2021)
- add test that validate_events_json return NA if jsonvalidate not installed

dlebauer committed:
Update CHANGELOG and NEWS

dlebauer committed:
Update testthat version requirement and documentation for validate_events_json

dlebauer committed:
Merge branch 'develop' into write_events_sipnet

AritraDey-Dev committed:
Update models/ed/tests/testthat/test-read_ET_files.R

Co-authored-by: Chris Black <chris@ckblack.org>

infotroph committed:
Merge pull request #3652 from AritraDey-Dev/tests/update-output-expectation

tests: use specific variable checks in read_ET_files

infotroph committed:
do not assume id is numeric when assigning str_ns

Commit:
Revert to using the new debias workflow.

Commit:
Update document.

Commit:
Update document.

Commit:
Update the runner script.

Commit:
Update the bias correction function.

Commit:
Update the usage of the bias correction workflow.

DongchenZ committed:
Merge branch 'PecanProject:develop' into SDA_data

Commit:
Update observation paths.

Commit:
Update namespace.

Commit:
Merge branch 'SDA_data' of https://github.com/DongchenZ/pecan into SDA_data

Commit:
Modify the test to match the change in the logger.message function.

Commit:
Change the default value of wrap in the logger message function.

Commit:
Revert back.

infotroph committed:
data.frame does not allow starting with null cols

AritraDey-Dev committed:
Merge pull request #8 from dlebauer/quarto-ensemble-sensivity

Migrating additional content from UI demos to Quarto demos

AritraDey-Dev committed:
migrate pecan outputs block from demo 1 to demo 2

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
add changes in demo 2

Signed-off-by: Aritra Dey <adey01027@gmail.com>

Commit:
Update runner script.

Commit:
Modify configuration.

infotroph committed:
Merge pull request #3656 from infotroph/siteID-refactor-rebase

Simpler str_ns handling

infotroph committed:
tests for GEFS functions

infotroph committed:
whitespace

infotroph committed:
simplify formatting logic, same result

infotroph committed:
move download_grid definition out to top level

infotroph committed:
simplify more

infotroph committed:
returns null

infotroph committed:
Merge pull request #3650 from infotroph/stics-fix

Stics Roxygen output

mdietze committed:
Merge pull request #3349 from infotroph/GEFS-fix

GEFS download fixes

infotroph committed:
date must be less than 4 days in past

infotroph committed:
and the other ones too

infotroph committed:
Merge pull request #3657 from infotroph/gefs-test-fix

fix GEFS test

infotroph committed:
Merge pull request #3623 from dlebauer/write_events_sipnet

Add write.events.SIPNET function

infotroph committed:
dbsync build assumes cwd

mdietze committed:
Merge pull request #3658 from infotroph/dbsync-fixes

dbsync tweaks

Commit:
Add the prefix argument to customize the desired output folder.

Commit:
Update configurations.

Commit:
Update the SDA runner script.

DongchenZ committed:
Merge branch 'PecanProject:develop' into SDA_data

Commit:
Merge branch 'SDA_data' of https://github.com/DongchenZ/pecan into SDA_data

infotroph committed:
Merge branch 'develop' into ensure-version-bump

infotroph committed:
dbsync has now changed too

infotroph committed:
Merge pull request #3631 from infotroph/ensure-version-bump

[CI] add check for packages needing version increment

infotroph committed:
fixes for checks failing in R 4.5

- Remove test for parsable empty string (behavior changed in units 1.0 and we dont actually care)
- add missing package ref to Rd link

infotroph committed:
wording improvements from code review

Co-authored-by: David LeBauer <dlebauer@arizona.edu>

infotroph committed:
Merge pull request #3659 from infotroph/units-emptystr-change

fixes for checks failing in R 4.5

infotroph committed:
Merge branch 'develop' into progressReport1

infotroph committed:
Merge branch 'develop' into mattykim-import

infotroph committed:
Merge pull request #3614 from sarahkanee/progressReport1

NDTI/EVI extraction function and example workflow

infotroph committed:
Merge pull request #3615 from sarahkanee/mattykim-import

Add Matt Kim's R scripts to data.remote/inst

AritraDey-Dev committed:
Update base/all/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/db/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/logger/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/qaqc/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/remote/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/settings/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/utils/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/visualization/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update base/workflow/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update models/basgra/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/uncertainty/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/rtm/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/priors/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/photosynthesis/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/meta.analysis/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/emulator/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/data.remote/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/data.mining/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/data.land/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/data.atmosphere/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/benchmark/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/assim.sequential/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/assim.batch/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update modules/allometry/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update models/maat/DESCRIPTION

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
apply chnages in description in model packages

Signed-off-by: Aritra Dey <adey01027@gmail.com>

dlebauer committed:
Apply suggestion from @dlebauer

infotroph committed:
Roxygen

dlebauer committed:
Merge pull request #3654 from infotroph/sa-multisite-hack

Partial fixes for multisite ensembles

infotroph committed:
Apply suggestions from code review

infotroph committed:
typo

infotroph committed:
typo again 🤦

infotroph committed:
doc, ::, whitespace

dlebauer committed:
- Remove description of trait data and meta-analysis from uncertainty.qmd demo
- Enable auto-numbered and cross-referenced sections

dlebauer committed:
add self as co-author

infotroph committed:
Merge pull request #3507 from AritraDey-Dev/keyword-for-packages

 enhance X-schema.org-keywords in DESCRIPTION files

AritraDey-Dev committed:
Merge pull request #10 from dlebauer/uncertainty_demo_update

Minor changes to Demo 2

infotroph committed:
Merge branch 'develop' into docker-use-arm-45

infotroph committed:
Merge pull request #3655 from infotroph/docker-use-arm-45

[docker] Add ARM support where available from upstream

AritraDey-Dev committed:
fix pft outdir path

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
Update CHANGELOG.md

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
Update documentation/tutorials/Demo_02_Uncertainty_Analysis/uncertainty.qmd

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
fix: output directory structure

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
Update .github/workflows/render-quarto.yml

Co-authored-by: Chris Black <chris@ckblack.org>

Commit:
Bug fixes for the indexing error.

Commit:
Fix the issue when X is a data frame.

Commit:
Add the adjustment to the analysis block function.

Commit:
Add the adjustment to the function.

Commit:
Add the residual lag to the SDA ML debiasing workflow.

Commit:
Use pre-defined soil depth for the soil physics calculation.

Commit:
Resolve conflicts.

Merge branch 'develop' of https://github.com/PecanProject/pecan into SDA_data

# Conflicts:
#	modules/uncertainty/R/ensemble.R

Commit:
Add comments for the reason of returning the list object.

Commit:
Fix the logic where we already generated analysis from the analysis block function.

infotroph committed:
Fix unit error in leafCSpWt calculation: Percent-to-fraction conversion now propagated back to LeafC

Co-authored-by: Qianyuli <liqianyu2528@hotmail.com>

infotroph committed:
Merge pull request #3664 from infotroph/sipnet-leaf-units-again

Fix unit error in leafCSpWt calculation

Commit:
Remove the ensemble.samples argument.

Commit:
Update documentation.

infotroph committed:
relicense PEcAn.uncertainty as BSD3 with agreement of all contributors

Commit:
Revert the previous comments and add the parameter generation if we don't have it.

DongchenZ committed:
Merge branch 'PecanProject:develop' into SDA_data

Commit:
Merge branch 'develop' of https://github.com/PecanProject/pecan into SDA_data

Commit:
Merge branch 'SDA_data' of https://github.com/DongchenZ/pecan into SDA_data

infotroph committed:
move DART to contrib/

infotroph committed:
remove other refs, wording tweak

infotroph committed:
remove flag with no effect after R 4.2

mdietze committed:
Update create_amerifluxLBL_drivers_for_maat.Rmd

When AmerifluxLBL download isn't working, this vignette stops the entire build, so making many of the code blocks "eval = FALSE"

infotroph committed:
Merge pull request #3669 from PecanProject/mdietze-patch-1

Update create_amerifluxLBL_drivers_for_maat.Rmd

infotroph committed:
Merge branch 'develop' into drop-check-flag

infotroph committed:
Merge branch 'develop' into move-dart

mdietze committed:
Merge branch 'develop' into relicense-uncertainty

mdietze committed:
Merge pull request #3667 from infotroph/drop-check-flag

remove flag with no effect after R 4.2

mdietze committed:
Merge pull request #3666 from infotroph/move-dart

Move DART to contrib/, clarify license

mdietze committed:
Merge branch 'develop' into SDA_data

mdietze committed:
Merge pull request #3634 from DongchenZ/SDA_data

Implement the `join_inputs_design` into the current SDA workflows.

infotroph committed:
Merge branch 'develop' into relicense-uncertainty

dlebauer committed:
Merge pull request #3665 from infotroph/relicense-uncertainty

relicense PEcAn.uncertainty as BSD-3

dlebauer committed:
Merge pull request #3601 from harshagr70/render_get.trait.data

Add flat-file bypass for get.trait.data()

AritraDey-Dev committed:
Update CHANGELOG.md

Co-authored-by: Chris Black <chris@ckblack.org>

AritraDey-Dev committed:
allow multi variable outputs for ensemble analysis

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
change naming for outputs

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
fix: add missing newlines to DESCRIPTION files to resolve warnings

Signed-off-by: Aritra Dey <adey01027@gmail.com>

infotroph committed:
Merge pull request #3670 from AritraDey-Dev/fix/warnings-for-description

fix: resolve DESCRIPTION file warnings by adding missing newlines

AritraDey-Dev committed:
Fix Makefile clean rule by removing wrong ../book/ directory

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
test(db): fix fancy_scientific test by comparing deparsed expressions

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
remove deploy rule from Makefile

Signed-off-by: Aritra Dey <adey01027@gmail.com>

infotroph committed:
Merge pull request #3671 from AritraDey-Dev/improve-make-clean

Correct clean rule in Makefile

infotroph committed:
Merge pull request #3672 from AritraDey-Dev/fix/db-tests

test(db): fix fancy_scientific test by comparing deparsed expressions

infotroph committed:
update pr.dens tests for testthat 3.3.0

dlebauer committed:
Merge pull request #3570 from AritraDey-Dev/quarto-ensemble-sensivity

Demo 2 Quarto notebook tutorial for PEcAn uncertainty analysis

AritraDey-Dev committed:
Merge branch 'develop' into ci/uncertainty-notebook

AritraDey-Dev committed:
style: fix R code formatting in uncertainty.qmd

Signed-off-by: Aritra Dey <adey01027@gmail.com>

infotroph committed:
Merge pull request #3675 from infotroph/remove-is_equal

update pr.dens tests for testthat 3.3.0

infotroph committed:
Merge pull request #3638 from AritraDey-Dev/ci/uncertainty-notebook

CI:  Demo 2 notebook rendering

Commit:
Update phenology parameters when there are NAs for specific site-years

infotroph committed:
Merge pull request #3680 from Qianyuxuan/phenology_update

Update phenological parameters when there are NAs for specific site-years using across-year means

infotroph committed:
ensure_version_bump: default to tag most recent by date, not by commit count

This works better in shallow clones (e.g. on CI) because you only need the tag commits (`git fetch --tags --depth=1`); previous `git describe` default needed the full history from HEAD to tag.

Note that neither of these is guaranteed to be the tag with the semantically most recent version number; for that we'd want --sort="v:refname" along with being sure our tags stick to one parseable versioning system.

AritraDey-Dev committed:
build(docs): misc cleanups & add serve rule to book_Makefile

Signed-off-by: Aritra Dey <adey01027@gmail.com>

AritraDey-Dev committed:
remove serve rule from Makefile

Signed-off-by: Aritra Dey <adey01027@gmail.com>

infotroph committed:
Merge pull request #3685 from AritraDey-Dev/cleanup-in-makefile

build(docs): minor cleanups in book_source/Makefile

infotroph committed:
Merge pull request #3686 from infotroph/fix-ensure-version-bump

ensure_version_bump: Default to most recent tag by date

infotroph committed:
version bumps and NEWS tweaks for model packages

infotroph committed:
testthat

dlebauer committed:
Merge pull request #3704 from infotroph/model-pkg-version-bump

version bump for model packages

infotroph committed:
bump versions for modules and base

infotroph committed:
missed one

divine7022 committed:
update CHANGELOG.md

divine7022 committed:
update roxy

divine7022 committed:
update roxy

divine7022 committed:
update run.write.configs.Rd

divine7022 committed:
runModule.run.write.configs.Rd

divine7022 committed:
update roxy

divine7022 committed:
update write.ensemble.configs.Rd

divine7022 committed:
add input_design documentation to book

divine7022 committed:
made message clear

divine7022 committed:
add input_design row validation

divine7022 committed:
add input_design row validation to runModule.run.write.configs

divine7022 committed:
update roxy

divine7022 committed:
update roxy

divine7022 committed:
update the documenation

infotroph committed:
pecan.all version follows pecan version

divine7022 committed:
drop reduendent info

infotroph committed:
Merge pull request #3715 from infotroph/v110-version-bump

bump versions for modules and base

infotroph committed:
fix release date in csv

infotroph committed:
version in web/common.php

infotroph committed:
Apply suggestions from code review

infotroph committed:
Merge pull request #3716 from divine7022/docs-joint-input-design

add input design documentation to book source

infotroph committed:
date fix

infotroph commented on modules/allometry/DESCRIPTION:
```diff
@@ -1,12 +1,14 @@
 Type: Package
 Title: PEcAn Allometry Functions
-Version: 1.7.4
+Version: 1.7.4.9000
```
Missed one!
```suggestion
Version: 1.7.5
```

infotroph commented on base/all/data/pecan_releases.csv:
```diff
@@ -33,3 +33,4 @@
 "v1.7.2",2021-10-08,1.7.2
 "v1.8.0",2024-07-12,1.8.0
 "v1.9.0",2025-05-25,1.9.0
+"v1.10.0",2025-12-16,1.10.0
```
```suggestion
"v1.10.0",2025-12-17,1.10.0
```

infotroph committed:
Apply suggestions from code review

infotroph committed:
date fix

infotroph committed:
date fix

S1DDHEY committed:
Adding Notebook-based tutorials section in README

Added notebook-based tutorials section with links and marked the web interface demo as legacy.

S1DDHEY committed:
fixed broken links and grammar in README.md

mdietze committed:
Merge pull request #3730 from S1DDHEY/fix/README.md-v1.10.0

Docs: update README to emphasize notebook tutorials

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

infotroph committed:
date

robkooper committed:
fix for ED and docker

docker.sh now mimics the github action closer

mdietze committed:
Merge pull request #3755 from PecanProject/docker-ed-fix

fix for ED and docker

infotroph committed:
date
